File: 卡码/代码随想录/C++/动态规划/dp/02-背包问题/src/02-完全背包/LeetCode_0322.cpp
//322. 零钱兑换
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
	int coinChange(vector<int>& coins, int amount) {
		vector<int> dp(amount+1,INT_MAX-2);
		dp[0]=0;
		for(int coin:coins){
			for(int j=coin;j<=amount;j++){
				dp[j]=min(dp[j],dp[j-coin]+1);
			}
		}
		if(dp[amount]==INT_MAX-1) return -1;
		return dp[amount];
	}
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/1/09-动态规划/Java/dp/src/main/java/com/kurumi/dp/_02_背包问题/_02_完全背包/LeetCode_0322.java
//322. 零钱兑换
package com.kurumi.dp._02_背包问题._02_完全背包;

import java.util.Arrays;

public class LeetCode_0322 {

}
class Solution {
  public int coinChange(int[] coins, int amount) {
    int[] dp=new int[amount+1];
    Arrays.fill(dp, Integer.MAX_VALUE-1);
    dp[0]=0;
    for(int coin:coins){
      for(int j=coin;j<=amount;j++){
        dp[j]=Integer.min(dp[j], dp[j-coin]+1);
      }
    }
    return dp[amount] == Integer.MAX_VALUE - 1 ? -1 : dp[amount];
  }
}
